Remotion/Web/Development.WebTesting/DownloadInfrastructure/DownloadedFileFinder.cs

```diff
@@ -171,23 +186,31 @@ private List<string> GetNewFiles (IEnumerable<string> filesInDownloadDirectoryBe
           .ToList();
     }
 
-    private bool TryGetFileInformation (string fileName, out FileInformationTuple fileInformation)
+    private bool TryGetFileInformations (IEnumerable<string> fileNames, out IDictionary<string, FileInformationTuple> fileInformations)
```

Comment:
Given that the dictionary is never null and the only result is an empty or a filled dictionary, this feels like it violates the intent of the try-pattern. I see how it helps with the conditions but it makes the code harder to reason over. For instance, when looking at this method alone, it's not clear why this isn't done as a simple Get-method. 

I haven't played with the code, but I think it can be simplified more / can be made easier to read by further restructuring. 

Caveat: Where does the idea about a complete rewrite factor in to all this?

Reply:
> Caveat: Where does the idea about a complete rewrite factor in to all this?

@MichaelKetting It doesn't. This PR fixes the current problem with "minimal effort" as a rewrite is planned anyway, as we talked about ~a week ago. Please let me know if the rewrite should happen immediately.

Reply:
@iam-mholle Mis-communication: meant, if we change this to GetFileInformations instead of Try..., is this just busiwork until we do the rework or is it a helpful code cleanup?

Reply:
I don't think it would make the code more understandable, rather increase the complexity by adding more nesting as the condition could no longer be written the way it is written now.

I actually also don't think it necessarily violates the `Try...` pattern (in my mind at least 🤷🏼‍♂️) as the boolean return value only indicates if the operation was successful, and that the `out` parameter value can be used if the return value is `true`. If the value is `false`, the `out` parameter value could be `null`, but could also be an instance of some type. Either way, it shouldn't be used from a semantic point of view.

Reply:
@iam-mholle okay, and once we actually do the rewrite, all this wonderful stuff will be deleted anyway and re-written with some other patterns, correct?

Reply:
> @iam-mholle okay, and once we actually do the rewrite, all this wonderful stuff will be deleted anyway and re-written with some other patterns, correct?

That's what I would expect.

Reply:
Okay, then please add a ticket for the rewrite, add a todo to the class do document that this will be rewritten and we're done.